io/src/main/scala/fs2/io/tcp/Socket.scala

```diff
@@ -147,67 +140,63 @@ protected[tcp] object Socket {
       implicit AG: AsynchronousChannelGroup,
       F: ConcurrentEffect[F],
       timer: Timer[F]
-  ): Stream[F, Either[InetSocketAddress, Stream[F, Socket[F]]]] =
-    Stream.suspend {
-
-      def setup: F[AsynchronousServerSocketChannel] = F.delay {
-        val ch = AsynchronousChannelProvider
-          .provider()
-          .openAsynchronousServerSocketChannel(AG)
-        ch.setOption[java.lang.Boolean](StandardSocketOptions.SO_REUSEADDR, reuseAddress)
-        ch.setOption[Integer](StandardSocketOptions.SO_RCVBUF, receiveBufferSize)
-        ch.bind(address)
-        ch
-      }
-
-      def cleanup(sch: AsynchronousServerSocketChannel): F[Unit] = F.delay {
-        if (sch.isOpen) sch.close()
-      }
+  ): Stream[F, Either[InetSocketAddress, Stream[F, Socket[F]]]] = {
```

Comment:
I'm not sure whether this should be `Stream[F, Either[InetSocketAddress, Resource[F, Socket[F]]]]` or not

Reply:
Yeah good point, it should